### ISSUE TITLE
2.4.6-p3 support

### ISSUE THREAD

**Issue by renvisk** [magento/meta-for-magento2#41]
Magento released security patches that fix some severe issues ([seen here](https://helpx.adobe.com/security/products/magento/apsb23-50.html)). Unfortunately it seems to conflict with the meta-for-magento2 extension, because of strict composer restrictions. 

### Preconditions (*)

1."magento/product-community-edition": "2.4.6-p3",
2."meta/meta-for-magento2": "*"

### Steps to reproduce (*)

1.Try to require the package for magento 2.4.6-p3.
2. Fail doing so because the package restricted to 2.4.6-p2

### Expected result (*)

1. Composer install runs successfully

### Actual result (*)

```
`Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Root composer.json requires magento/product-community-edition 2.4.6-p3 -> satisfiable by magento/product-community-edition[2.4.6-p3].
    - meta/module-sales[1.2.0, ..., 1.2.1] require magento/module-shipping >=100.4.4 <=100.4.6-p2 -> satisfiable by magento/module-shipping[100.4.4, 100.4.5, 100.4.6, 100.4.6-p2].
    - You can only install one version of a package, so only one of these can be installed: magento/module-shipping[100.0.2, ..., 100.4.6-p3].
    - meta/meta-for-magento2[1.2.0, ..., 1.2.1] require meta/module-sales 1.2.* -> satisfiable by meta/module-sales[1.2.0, 1.2.1].
    - magento/product-community-edition 2.4.6-p3 requires magento/module-shipping 100.4.6-p3 -> satisfiable by magento/module-shipping[100.4.6-p3].
    - Root composer.json requires meta/meta-for-magento2 ^1.2 -> satisfiable by meta/meta-for-magento2[1.2.0, 1.2.1].
```

**Comment by magentix** [comment]
This issue prevents the installation of the latest security release.

I'm not really sure that restrictions on security releases are necessary. Why not just use Caret Version Range?

```
"magento/framework": "^103.0.4",
```

Instead of:

```
"magento/framework": ">=103.0.4 <=103.0.6-p2",
```

**Comment by strygo** [comment]
This issue is now fixed in v1.2.2 - thank you for flagging this!
